File: src/main/java/play/pluv/playlist/domain/PlayListMusic.java
package play.pluv.playlist.domain;

import java.util.List;
import java.util.Optional;
import lombok.Builder;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.ToString;

@RequiredArgsConstructor
@Builder
@ToString
public class PlayListMusic {

  @Getter
  private final String title;
  @Getter
  private final List<String> artistNames;
  //Nullable
  private final String isrcCode;
  @Getter
  private final String imageUrl;

  public Optional<String> getIsrcCode() {
    return Optional.ofNullable(isrcCode);
  }
}

File: src/main/kotlin/play/pluv/playlist/domain/PlayListMusic.kt
package play.pluv.playlist.domain

import java.util.*

//TODO: Optional 제거
data class PlayListMusic(
    val title: String,
    val artistNames: List<String>,
    val imageUrl: String,
    @get:JvmName("getNullableIsrcCode")
    val isrcCode: String?,
) {

    fun getIsrcCode(): Optional<String> = Optional.ofNullable(isrcCode)
}